### PR TITLE
Revert "CakeFixtureManager: Truncate tables only if not dropped."

### PR DESCRIPTION
**Pull request by markstory**
Reverts cakephp/cakephp#9368. This change caused regressions in applications.

Refs #9452

### CODE REVIEW

File: lib/Cake/TestSuite/Fixture/CakeFixtureManager.php [hunk 1]
```diff
@@ -98,7 +98,7 @@ protected function _initDb() {
 /**
  * Parse the fixture path included in test cases, to get the fixture class name, and the
  * real fixture path including sub-directories
- *
+ *
```
Comment (dereuromark) [inline lib/Cake/TestSuite/Fixture/CakeFixtureManager.php]: some noise

File: lib/Cake/TestSuite/Fixture/CakeFixtureManager.php [hunk 2]
```diff
@@ -218,7 +218,7 @@ public function load(CakeTestCase $test) {
 		if (empty($test->fixtures)) {
 			return;
 		}
-		$fixtures = array_unique($test->fixtures);
+		$fixtures = $test->fixtures;
```
Comment (dereuromark) [inline lib/Cake/TestSuite/Fixture/CakeFixtureManager.php]: that should stay IMO.

**Comment by dereuromark** [comment]
Any idea why this would cause regressions and why we cannot fix?

**Comment by chinpei215** [comment]
@dereuromark Probably, [this line](https://github.com/cakephp/cakephp/blob/d52b9fc826edd5e30133bbbdb9a8ecb1bf8ee504/lib/Cake/TestSuite/Fixture/CakeFixtureManager.php#L194). _setupTable does nothing if the specified table has been already created. In this case, loadSingle doesn't truncate existing records, but [insert](https://github.com/cakephp/cakephp/blob/d52b9fc826edd5e30133bbbdb9a8ecb1bf8ee504/lib/Cake/TestSuite/Fixture/CakeFixtureManager.php#L282).

**Comment by dereuromark** [comment]
@chinpei215 Do you think we can fix it or is it better to revert it and not having the speed advantage here?

**Comment by chinpei215** [comment]
@dereuromark Probably, if _setupTable() returns boolean (true = new table was created / false = existing table was reused), the caller can judge whether truncate() should be called or not.

File: lib/Cake/TestSuite/Fixture/CakeFixtureManager.php (hunk 1 above)
Comment (markstory) [inline lib/Cake/TestSuite/Fixture/CakeFixtureManager.php]: I simply reverted the changes from within github.

**Pull request merged**